### PR TITLE
Release azure cosmos 4.47.0 and azure cosmos spark 4.20.0

### PR DESCRIPTION
**Pull request by jeet1995**
Release branch with version updates for `azure-cosmos` and `azure-cosmos-spark`

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
-### 4.47.0-beta.1 (Unreleased)
+### 4.47.0 (2023-06-23)
 
 #### Features Added
 * Added the capability to specify region switch hints through `CosmosClientBuilder#setSessionRetryOptions` for optimizing retries for `READ_SESSION_NOT_AVAILABLE` errors. - See [PR 35292](https://github.com/Azure/azure-sdk-for-java/pull/35292)
-
-#### Breaking Changes
+* Added API to exclude regions on request options which helps avoid a regions from preferred regions for the request. - See [PR 35166](https://github.com/Azure/azure-sdk-for-java/pull/35166)
+* Added API for providing an availability strategy to improve availability when end-end timeout is specified. - See [PR 35166](https://github.com/Azure/azure-sdk-for-java/pull/35166)
+* Added Threshold based availability strategy - see PR for details. - See [PR 35166](https://github.com/Azure/azure-sdk-for-java/pull/35166)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:9]:
```suggestion
* Added Threshold based availability strategy. - See [PR 35166](https://github.com/Azure/azure-sdk-for-java/pull/35166)
```

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: Fixed in next iteration.

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/README.md [hunk 2]
```diff
@@ -28,7 +28,8 @@ https://github.com/Azure/azure-sdk-for-java/issues/new
 
 #### azure-cosmos-spark_3-1_2-12
 | Connector    | Supported Spark Versions | Minimum Java Version | Supported Scala Versions | Supported Databricks Runtimes |
-|--------------| ------------------------ | -------------------- | -----------------------  | ----------------------------- |
+|--------------|--------------------------|----------------------|--------------------------|-------------------------------|
+| 4.20.0       | 3.1.1 - 3.1.2            | 8                    | 2.12                     | 11.\*                         |
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/README.md:32]: Supported Databricks Runtimes  -> should be align with other versions

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/README.md [hunk 3]
```diff
@@ -73,7 +74,8 @@ https://github.com/Azure/azure-sdk-for-java/issues/new
 
 #### azure-cosmos-spark_3-2_2-12
 | Connector | Supported Spark Versions | Minimum Java Version | Supported Scala Versions | Supported Databricks Runtimes |
-|-----------| ------------------------ | -------------------- | -----------------------  | ----------------------------- |
+|-----------|--------------------------|----------------------|--------------------------|-------------------------------|
+| 4.20.0    | 3.2.0 - 3.2.1            | 8                    | 2.12                     | 11.\*                         |
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/README.md:78]: `Supported Databricks Runtimes `-> should be align with other versions

File: sdk/cosmos/azure-cosmos-spark_3-3_2-12/README.md [hunk 4]
```diff
@@ -39,7 +40,8 @@ https://github.com/Azure/azure-sdk-for-java/issues/new
 
 #### azure-cosmos-spark_3-2_2-12
 | Connector | Supported Spark Versions | Minimum Java Version | Supported Scala Versions | Supported Databricks Runtimes |
-|-----------| ------------------------ | -------------------- | -----------------------  | ----------------------------- |
+|-----------|--------------------------|----------------------|--------------------------|-------------------------------|
+| 4.20.0    | 3.2.0 - 3.2.1            | 8                    | 2.12                     | 11.\*                         |
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-3_2-12/README.md:44]: `Supported Databricks Runtimes` -> should be align with other versions

File: sdk/cosmos/azure-cosmos-spark_3-3_2-12/README.md [hunk 5]
```diff
@@ -69,7 +71,8 @@ https://github.com/Azure/azure-sdk-for-java/issues/new
 
 #### azure-cosmos-spark_3-1_2-12
 | Connector    | Supported Spark Versions | Minimum Java Version | Supported Scala Versions | Supported Databricks Runtimes |
-|--------------| ------------------------ | -------------------- | -----------------------  | ----------------------------- |
+|--------------|--------------------------|----------------------|--------------------------|-------------------------------|
+| 4.20.0       | 3.2.0 - 3.2.1            | 8                    | 2.12                     | 11.\*                         |
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-3_2-12/README.md:75]: `Supported Databricks Runtimes` -> should be align with other versions

File: sdk/cosmos/azure-cosmos-spark_3-3_2-12/README.md [hunk 6]
```diff
@@ -27,7 +27,8 @@ https://github.com/Azure/azure-sdk-for-java/issues/new
 
 #### azure-cosmos-spark_3-3_2-12
 | Connector | Supported Spark Versions | Minimum Java Version | Supported Scala Versions | Supported Databricks Runtimes |
-|-----------|--------------------------| -------------------- | -----------------------  |-------------------------------|
+|-----------|--------------------------|----------------------|--------------------------|-------------------------------|
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-3_2-12/README.md:30]: will also need to update the version referenced in quick-start.md

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/SessionRetryOptions.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+     * NOTES:
+     * <ul>
+     *     <li>{@code null} values are not allowed</li>
+     * </ul>
+     *
+     * @param regionSwitchHint The region switch hint
+     * @return This instance of {@link SessionRetryOptions}
+     * */
+    public SessionRetryOptions setRegionSwitchHint(CosmosRegionSwitchHint regionSwitchHint) {
+        this.regionSwitchHint = regionSwitchHint;
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/SessionRetryOptions.java:38]: We need to check the nulls here.

**Review by kushagraThapar — CHANGES_REQUESTED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/SessionRetryOptions.java (hunk 7 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/SessionRetryOptions.java:41]: Fixed in next iteration.

**Review by kushagraThapar — APPROVED** [review]

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]
As discussed in Teams - without the SessionRetryOptionsBuilder the region switch hint becomes mutable - we need to double-check with Bhaskar that this is intentional and acceptable and that we have proper test coverage for it. My preference would be to keep this immutable

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/SessionRetryOptionsBuilder.java [hunk 8]
```diff
@@ -1,15 +1,12 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/SessionRetryOptionsBuilder.java:1]: Did we remove this by mistake? If so, pls put it back.

Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/SessionRetryOptionsBuilder.java]: Fixed in next iteration.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM now

**Review by kushagraThapar — APPROVED** [review]

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**